### PR TITLE
Embedding dac and other debugging resources into singlefilehost

### PR DESCRIPTION
**Pull request by VSadov**

### CODE REVIEW

File: src/coreclr/dlls/mscoree/coreclr/CMakeLists.txt [hunk 1]
```diff
@@ -204,16 +204,31 @@ if(CLR_CMAKE_TARGET_WIN32)
...
+        # make CLRDEBUGINFO resource and inject into coreclr and singlefilehost
         COMMAND GenClrDebugResource /dac:$<TARGET_FILE:mscordaccore> /dbi:$<TARGET_FILE:mscordbi> /sku:onecoreclr /out:${CMAKE_CURRENT_BINARY_DIR}/clrDebugResource.bin
         COMMAND InjectResource /bin:${CMAKE_CURRENT_BINARY_DIR}/clrDebugResource.bin /dll:$<TARGET_FILE:coreclr> /name:CLRDEBUGINFO
+        COMMAND InjectResource /bin:${CMAKE_CURRENT_BINARY_DIR}/clrDebugResource.bin /dll:$<TARGET_FILE:singlefilehost> /name:CLRDEBUGINFO
+
+        # inject MINIDUMP_AUXILIARY_PROVIDER into coreclr
         COMMAND InjectResource /bin:${CMAKE_CURRENT_SOURCE_DIR}/dump_helper_resource.bin /dll:$<TARGET_FILE:coreclr> /name:MINIDUMP_AUXILIARY_PROVIDER
+
+        # inject mscordaccore.dll as MINIDUMP_AUXILIARY_EMBEDDED_MODULE resource into singlefilehost
+        COMMAND InjectResource /bin:$<TARGET_FILE:coreclr> /dll:$<TARGET_FILE:singlefilehost> /name:MINIDUMP_AUXILIARY_EMBEDDED_MODULE
```
Comment (hoyosjs) [inline src/coreclr/dlls/mscoree/coreclr/CMakeLists.txt:228]: Just this part is wrong,  `mscordaccore` would be the TARGET_FILE. But at this point of time it is not signed.

Comment (VSadov) [reply src/coreclr/dlls/mscoree/coreclr/CMakeLists.txt]: Right. copy/paste error. I was wondering why the file so big.

Comment (VSadov) [reply src/coreclr/dlls/mscoree/coreclr/CMakeLists.txt]: Should be picking it up in installer partition now.

File: src/native/corehost/build.cmd [hunk 2]
```diff
@@ -186,6 +186,9 @@ if "%__RuntimeFlavor%" NEQ "Mono" (
     echo Copying "%__CoreClrArtifacts%\corehost\PDB\singlefilehost.pdb"  "%__CMakeBinDir%/corehost/PDB/"
     copy /B /Y "%__CoreClrArtifacts%\corehost\PDB\singlefilehost.pdb"  "%__CMakeBinDir%/corehost/PDB/"
 
+    echo Embedding "%__CoreClrArtifacts%\mscordaccore.dll" into "%__CMakeBinDir%\corehost\singlefilehost.exe"
+    "%__CoreClrArtifacts%\dactabletools\InjectResource.exe" /bin:"%__CoreClrArtifacts%\mscordaccore.dll" /dll:"%__CMakeBinDir%\corehost\singlefilehost.exe" /name:MINIDUMP_AUXILIARY_EMBEDDED_MODULE
```
Comment (hoyosjs) [inline src/native/corehost/build.cmd:190]: The only thing I'd love is to be able to verify here that what we are embedding is actually 200-signed.

**Review by hoyosjs — APPROVED** [review]
Thanks! This was quick

File: src/native/corehost/build.cmd (hunk 2 above)
Comment (hoyosjs) [inline src/native/corehost/build.cmd:190]:
```suggestion
    "%__CoreClrArtifacts%\dactabletools\InjectResource.exe" /bin:"%__CoreClrArtifacts%\mscordaccore.dll" /dll:"%__CMakeBinDir%\corehost\singlefilehost.exe" /name:MINIDUMP_EMBEDDED_AUXILIARY_PROVIDER
```

Comment (VSadov) [reply src/native/corehost/build.cmd]: That would be nice. Do we have tools examples for that?
We can also do it as a separate change if this PR could unblock further work.

Comment (VSadov) [reply src/native/corehost/build.cmd]: I assume something verifies mscordac.dll signature?

Comment (hoyosjs) [reply src/native/corehost/build.cmd]: Nothing really verifies it within the repo.

Comment (VSadov) [reply src/native/corehost/build.cmd]: I think we should merge this then and add verification when/if we figure how to verify.

**Comment by VSadov** [comment]
Thanks!!

**Pull request merged**